angular/src/directives/navigation/ion-tabs.ts

```diff
@@ -137,4 +164,43 @@ export class IonTabs {
   getSelected(): string | undefined {
     return this.outlet.getActiveStackId();
   }
+
+  /**
+   * Detects changes to the slot attribute of the tab bar.
+   *
+   * If the slot attribute has changed, then the tab bar
+   * should be relocated to the new slot position.
+   */
+  private detectSlotChanges(): void {
+    this.tabBars.forEach((tabBar: any) => {
+      // el is a protected attribute from the generated component wrapper
+      const currentSlot = tabBar.el.getAttribute('slot');
+
+      if (currentSlot !== this.tabBarSlot) {
+        this.tabBarSlot = currentSlot;
+        this.relocateTabBar();
+      }
+    });
+  }
+
+  /**
+   * Relocates the tab bar to the new slot position.
+   */
+  private relocateTabBar(): void {
+    const tabBar = (this.tabBar as any).el as HTMLElement;
```

Comment:
We have options here regarding the `any` type (and the `any` on L175). Our automatic generated proxies have the `el` property as protected. To make the types valid, we would need to disable automatically generating the proxy and manually wrap it with a `public` field member. 

I don't think the effort of maintaining the proxy is worth the type efficiency here.

Reply:
That makes sense. It might be a good idea to include a comment for the reasoning to keep it as type `any`.